src/Builder/DatagridBuilder.php

```diff
@@ -71,8 +72,12 @@ public function fixFieldDescription(AdminInterface $admin, FieldDescriptionInter
         // set default values
         $fieldDescription->setAdmin($admin);
 
-        if ($admin->getModelManager()->hasMetadata($admin->getClass())) {
-            [$metadata, $lastPropertyName, $parentAssociationMappings] = $admin->getModelManager()->getParentMetadataForProperty($admin->getClass(), $fieldDescription->getName());
+        $modelManager = $admin->getModelManager();
+
+        \assert($modelManager instanceof ModelManager);
```

Comment:
FYI, this will be always true in next major.

https://github.com/sonata-project/SonataAdminBundle/pull/6539/files#diff-0850515af24b81ce716f53b8eb0b71fae91dcbdcc88b28d95a83886504e1c8f4R1950

Reply:
This is to make sure the `ModelManager` is the one from this bundle, it is not about being `nullable`.

Reply:
I thought again and if some supplementary methods are needed, shouldn't an Interface be added instead ; in order to allow someone to use his custom ModelManager (especially since it will be final in next major) ?

Reply:
My idea was to see a way to add https://github.com/sonata-project/SonataDoctrineMongoDBAdminBundle/pull/461 so there is no need for these checks (if it can't be done in a bc way, in next major).